src/framework/linalg/eigensystem.hpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2018, 2019, 2020.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+#ifndef _aer_framework_linalg_eigensystem_hpp_
+#define _aer_framework_linalg_eigensystem_hpp_
+
+#include <type_traits>
+#include "framework/blas_protos.hpp"
+#include "framework/matrix.hpp"
+
+
+/**
+ * Returns the eigenvalues and eigenvectors
+ * of a Hermitian matrix.
+ * Uses the blas function ?heevx
+ * @param hermitian_matrix: The Hermitian matrix.
+ * @param eigenvalues: On output: the eignevalues of the matrix (input is overwritten)
+ * @param eigenvectors: On input: a copy of psd_matrix, on output: the eigenvectors of the matrix.
+ *
+ * @returns: void
+ */
+template <class T>
+void eigensystem_hermitian(const matrix<std::complex<T>>& hermitian_matrix,
+               /* out */ std::vector<T>& eigenvalues,
+               /* out */ matrix<std::complex<T>>& eigenvectors);
```

Comment:
Are the eigenvectors stored as the rows, or the columns of this matrix? This should be added to the documentation

Reply:
Columns. I'll add to the documentation.